src/NBitcoin/Crypto/ECDSASignature.cs

```diff
@@ -88,6 +88,20 @@ public ECDSASignature MakeCanonical()
                 return this;
         }
 
+        /// <summary>
+        /// Allow creation of signature with non-LowS for test purposes
+        /// </summary>
+        /// <remarks>Not to be used under normal circumstances</remarks>
+        public ECDSASignature MakeNonCanonical()
+        {
+            if (!this.IsLowS)
+            {
+                return this;
+            }
+            else
+                return new ECDSASignature(this.R, ECKey.CURVE_ORDER.Subtract(this.S));
```

Comment:
Did you check that this is correct?

Reply:
Yes. I found multiple sources that confirm that, for an ECDSA signature, the `R` and `S` values are both modulo `N` (aka the curve order). Further, a signature `(R, S)` is equivalent to `(R, -S mod N)`.

In other words there are always 2 valid `S` values, call them `S` and `S'`. Clearly `S + S' = N`, so there will always be one value greater than half the curve order (`N / 2`) and one less than this, unless both somehow happen to be `= N / 2`. The canonical signature is required to be the so-called 'low S' value, the one less than `N / 2`.

Therefore to get the other `S'` value (the complement) we calculate `S' = N - S`.

We can switch between the canonical and non-canonical form by calculating the complement of whichever representation we currently have in a signature.

Using `N + S` will give a valid signature too, but will not give the complement, as `(N + S) mod N = S`

Reply:
A small example of why `S + S' = N`:

`N = 7`
`S = 4`

`((N - S) % N)` = `3` <- `S'`, therefore `S` + `S'` = `7` = `N`
`((-S) % N)` = `3`

And `(N + S) mod N = S`:

`((N + S) % N)` = `4`

Reply:
Ok thanks very much for this explanation @zeptin